### ISSUE TITLE
AbstractPdf - ZendException font is not set

### ISSUE THREAD

**Issue by jchenevey** [magento/magento2#11743]
### Preconditions

1. Magento v2.2.0
2. PHP 7.0.22

### Steps to reproduce

1. Try to write a large multi-line block of text using `drawLineBlocks()` (my apologies for not providing a better example here - this came up during development of a custom module, which I am unable to include here)

### Expected result

1. PDF should render

### Actual result

1. A new Zend_Pdf_Exception is thrown, stating "Font has not been set"

The issue is caused by lines 1034-1036 in /vendor/magento/module-sales/Model/Order/Pdf/AbstractPdf.php.

```
foreach ($column['text'] as $part) {
    if ($this->y - $lineSpacing < 15) {
        $page = $this->newPage($pageSettings);
    }
...
```

This new page is created after the font has been determined (lines 1008-1025) and set for the line to be written. Therefore, at the time line 1057 is reached (below), the font for the new page has not been set.

`$page->drawText($part, $feed, $this->y - $top, 'UTF-8');`

The solution would be to re-set the page's font after creating the new page.

**Comment by magento-engcom-team** [comment]
@jchenevey, thank you for your report.
We've created internal ticket(s) MAGETWO-84054 to track progress on the issue.

**Comment by okorshenko** [comment]
Hi @jchenevey. Thank you for your report. 
The issue has been fixed in magento-engcom/magento2ce#1016 by @serhii-balko in 2.2-develop branch
Related commit(s):
 - 2629726dd4754b5fd50eebb4860794b7931fdd88
 - b326714be1661f917b23fea90576512b9c631f99
 - 5f4363b243fa0c297a8be2cab4c00f0d806b14c1
 - 9ce4730da51d91f1a1925f41b9ae6d573d9e96bd

The fix will be available with the upcoming patch release.
